In [ ]:
pip install folium

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2/2 [geopy]
Note: you may need to restart the kernel to use updated packages.


In [37]:
# Configurar pandas para mejor visualización
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
import json
import os
import sys
from pathlib import Path

import folium
from folium.plugins import HeatMap

warnings.filterwarnings('ignore')
# Configuraciones globales
pd.set_option('display.float_format', '{:,.2f}'.format)  # Con separador de miles
pd.set_option('display.max_columns', None)  # Mostrar todas las columnas
pd.set_option('display.width', 1000)  # Ancho de pantalla

print("✅ Librerías cargadas correctamente")

✅ Librerías cargadas correctamente


In [4]:
#Carga del dataset
# Agregar src al path (funciona desde cualquier ubicación)
# Buscar src/ hacia arriba
current = Path.cwd()
src_path = None

for parent in [current] + list(current.parents):
    if (parent / "src").exists() and (parent / "src" / "utils").exists():
        src_path = parent / "src"
        break

if src_path and str(src_path) not in sys.path:
    sys.path.insert(0, str(src_path))
    print(f"✅ src agregado al path: {src_path}")

# Importar DataLoader
from utils.data_loader import loader

# Listar datasets disponibles
loader.list_datasets("raw")

# Cargar el dataset
df = loader.load_csv("Motor_Vehicle_Collisions.csv", "raw")

if df is not None:
    print(f"\n✅ Dataset cargado exitosamente!")
else:
    print("\n❌ No se pudo cargar el dataset")


📊 Datasets en raw/:
  - IDB_INFRALATAM.csv (5.12 MB)
  - clientes.csv (1.40 MB)
  - transacciones_raw.csv (0.01 MB)
  - Motor_Vehicle_Collisions.csv (412.40 MB)
  - WB_BID.csv (0.22 MB)
  - temperaturas.csv (2.02 MB)
  - ventas.csv (3.73 MB)
  - mvcc_work.csv (453.44 MB)
  - products-10000.csv (1.53 MB)
  - organizations-10000.csv (1.33 MB)
  - 2025_manhattan (1).xlsx (1.70 MB)
  - 2025_manhattan (2).xlsx (1.70 MB)
✅ Cargando: /workspace/data/raw/Motor_Vehicle_Collisions.csv
📊 Shape: (2034019, 29)

✅ Dataset cargado exitosamente!


In [ ]:

df=df.sort_values('CRASH DATE', ascending=True)


,CRASH DATE,CRASH TIME,BOROUGH,ZIP CODE,LATITUDE,LONGITUDE,LOCATION,ON STREET NAME,CROSS STREET NAME,OFF STREET NAME,NUMBER OF PERSONS INJURED,NUMBER OF PERSONS KILLED,NUMBER OF PEDESTRIANS INJURED,NUMBER OF PEDESTRIANS KILLED,NUMBER OF CYCLIST INJURED,NUMBER OF CYCLIST KILLED,NUMBER OF MOTORIST INJURED,NUMBER OF MOTORIST KILLED,CONTRIBUTING FACTOR VEHICLE 1,CONTRIBUTING FACTOR VEHICLE 2,CONTRIBUTING FACTOR VEHICLE 3,CONTRIBUTING FACTOR VEHICLE 4,CONTRIBUTING FACTOR VEHICLE 5,COLLISION_ID,VEHICLE TYPE CODE 1,VEHICLE TYPE CODE 2,VEHICLE TYPE CODE 3,VEHICLE TYPE CODE 4,VEHICLE TYPE CODE 5
1825928,01/01/2013,14:45,BROOKLYN,11203,40.66,-73.92,"(40.6564831, -73.924076)",REMSEN AVENUE,EAST 57 STREET,NaN,0.00,0.00,0,0,0,0,0,0,Driver Inattention/Distraction,Unspecified,NaN,NaN,NaN,136740,OTHER,PASSENGER VEHICLE,NaN,NaN,NaN
1829642,01/01/2013,17:10,QUEENS,11414,40.66,-73.84,"(40.6641936, -73.8411164)",CROSS BAY BOULEVARD,156 AVENUE,NaN,0.00,0.00,0,0,0,0,0,0,Failure to Yield Right-of-Way,Unspecified,Unspecified,Unspecified,NaN,229499,PASSENGER VEHICLE,PASSENGER VEHICLE,SPORT UTILITY / STATION WAGON,SPORT UTILITY / STATION WAGON,NaN
1832669,01/01/2013,7:53,BROOKLYN,11203,40.65,-73.93,"(40.64632, -73.9298028)",UTICA AVENUE,BEVERLEY ROAD,NaN,0.00,0.00,0,0,0,0,0,0,Turning Improperly,Unspecified,NaN,NaN,NaN,136741,PASSENGER VEHICLE,PASSENGER VEHICLE,NaN,NaN,NaN
1832599,01/01/2013,22:00,QUEENS,11421,40.69,-73.85,"(40.6936625, -73.8521639)",WOODHAVEN BOULEVARD,JAMAICA AVENUE,NaN,1.00,0.00,1,0,0,0,0,0,Driver Inattention/Distraction,NaN,NaN,NaN,NaN,205108,SPORT UTILITY / STATION WAGON,NaN,NaN,NaN,NaN
1829027,01/01/2013,15:40,BROOKLYN,11207,40.67,-73.90,"(40.6685479, -73.9007828)",SUTTER AVENUE,SNEDIKER AVENUE,NaN,0.00,0.00,0,0,0,0,0,0,Unspecified,NaN,NaN,NaN,NaN,165521,SPORT UTILITY / STATION WAGON,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1972479,12/31/2022,22:15,BROOKLYN,"11,205.00",40.70,-73.97,"(40.696262, -73.973854)",NaN,NaN,143 PARK AVENUE,0.00,0.00,0,0,0,0,0,0,Unspecified,NaN,NaN,NaN,NaN,4594336,Sedan,NaN,NaN,NaN,NaN
1973271,12/31/2022,23:50,NaN,NaN,40.64,-73.88,"(40.638695, -73.87859)",BELT PARKWAY,NaN,NaN,10.00,0.00,0,0,0,0,10,0,Unsafe Lane Changing,Unspecified,Unspecified,NaN,NaN,4595351,Sedan,Sedan,Sedan,NaN,NaN
1972478,12/31/2022,23:30,NaN,NaN,40.68,-73.78,"(40.68315, -73.780304)",165 STREET,NaN,NaN,0.00,0.00,0,0,0,0,0,0,Failure to Yield Right-of-Way,Unspecified,NaN,NaN,NaN,4594383,Sedan,NaN,NaN,NaN,NaN
1972837,12/31/2022,20:05,NaN,NaN,40.69,-73.92,"(40.686497, -73.9168)",HANCOCK STREET,BROADWAY,NaN,1.00,0.00,0,0,0,0,1,0,Other Vehicular,Failure to Yield Right-of-Way,NaN,NaN,NaN,4595021,Ambulance,Sedan,NaN,NaN,NaN


In [ ]:
def resumen_accidente_por_ciudad(df):
    """
        Regresa un df con la ciudad o localidad dentro de nyc y el número de accidentes que ocurrieron 
        Resp:
        {
                {
                    localidad:QUEENS,
                    total_accidentes:150,
                    estatus:alto
                },
            {
                {
                    localidad:BROOKLYN,
                    total_accidentes:5,
                    estatus:BAJO
                }
            }
        }
    """
    # total de accidentes por localidad
    df_resumen=df['BOROUGH'].value_counts().reset_index()
    df_resumen.columns = ['localidad', 'total_accidentes']
    df_resumen['percentil'] = df_resumen['total_accidentes'].rank(pct=True) * 100

    #Establecemos el estatus para el mapa de calor
    condiciones=[
        df_resumen['percentil']<50,
        (df_resumen['percentil']>=50)  & (df_resumen['percentil']<75)
    ]

    valores=['BAJO','ALTO']
    df_resumen['estatus']=np.select(condiciones, valores, default='ALTO')

    return df_resumen

df_resumen=resumen_accidente_por_ciudad(df)
print(60*"=")
print("Resumen de total de accidentes por localidad")
total_accidentes=df_resumen['total_accidentes'].sum()
print(f'Total de accidentes en Nueva york:{total_accidentes}')
print(df_resumen.head())


Resumen de total de accidentes por localidad
Total de accidentes en Nueva york:1401232
       localidad  total_accidentes  percentil estatus
0       BROOKLYN            444936     100.00    ALTO
1         QUEENS            375361      80.00    ALTO
2      MANHATTAN            315278      60.00    ALTO
3          BRONX            206913      40.00    BAJO
4  STATEN ISLAND             58744      20.00    BAJO


2034019

***Encontramos un detalle interesante el total de accidentes por lacalidad es de 
1401232 pero el campo Borough tienen 31.11 % de nulos con respecto al total de registros 632787 nulos la suma de los dos da el total de registros que son de 2034019***

** Necesitamos imputar el resto de las localidades por no sabemos donde ocurrieron el 30% de accidentes y eso podria cambiar el mapa de calor

In [ ]:
#probamos la funcion de geo para localidad
